views-core/src/main/java/io/micronaut/views/ViewsFilter.java

```diff
@@ -88,42 +93,41 @@ public int getOrder() {
 
                         MediaType type = route.getValue(Produces.class, MediaType.class)
                                 .orElse((route.getValue(View.class).isPresent() || body instanceof ModelAndView) ? MediaType.TEXT_HTML_TYPE : MediaType.APPLICATION_JSON_TYPE);
-                        Optional<ViewsRenderer> optionalViewsRenderer = beanLocator.findBean(ViewsRenderer.class,
+                        Collection<ViewsRenderer> viewsRenderersCollection = beanLocator.getBeansOfType(ViewsRenderer.class,
```

Comment:
The ViewFilter is already a bit of a mess in terms of complexity IMHO, I would humbly suggest we consider extracting the view locator logic into an interface (ViewRendererLocator maybe?) with the default implementation being the one that just uses the Bean Locator. That would give users the ability to swap that bean with whatever logic fits their needs. I have 50% of that implemented in a branch that I can clean up and push up if would make sense.

Reply:
Extracting that logic into an interface would also allow us to close this feature request: https://github.com/micronaut-projects/micronaut-views/issues/56

Reply:
I am not against moving the `ViewsRenderer::exists` to a different interface but that could go into a different PR. Moreover, it would be a breaking change. 

I you don't see any issues with this PR I would like to release 2.4.0 containing this feature.